Docs: highlight structure vision (#342978)
NixOS/nixpkgs: doc/README.md

This directory houses the sources files for the Nixpkgs reference manual.

> [!IMPORTANT]
> We are actively restructuring our documentation to follow the [Diátaxis framework](https://diataxis.fr/)
>

> > Going forward, this folder should only contain **[reference](https://nix.dev/contributing/documentation/diataxis#reference) documentation**.

Reviewer: ```suggestion
> Going forward, this directory should **only** contain [reference documentation](https://nix.dev/contributing/documentation/diataxis#reference).
```

> For tutorials, guides and explanations, contribute to <https://nix.dev/> instead.
>
> We are actively working to generate **all** reference documentation from the [doc-comments](https://github.com/NixOS/rfcs/blob/master/rfcs/0145-doc-strings.md) present in code.
> This also provides the benefit of using `:doc` in the `nix repl` to view reference documentation locally on the fly.

> For reference documentation prefer Markdown files and code comments in the source code if supported yet.

Reviewer: This has changed meaning, but I don't understand why.

```suggestion
For documentation only relevant for contributors, use Markdown files next to the source and regular code comments.
```

Author: Doc-comments are not only relevant for contributors. They are reference documentation which is usually public.

Markdown files next to .nix files could be used if doc-comments directly in the nix source would be too big.
The nixdoc support is ready for this: https://github.com/nix-community/nixdoc/pull/114
But we didn't merge it. I guess we have a bootstrapping problem then.

Reviewer: But then I don't understand the new sentence at all. What else should people use for documentation except markdown and source comments? The original sentence was a hint for separating end-user and contributor documentation, so it's unrelated to the highlighted note on using doc-comments, which is why I don't think it should even be touched.

Author: Okay seems fine

> > We'are actively soliciting feedback for missing support. If your case is not supported open an issue starting with `Doc:`

Reviewer: I'm not sure what this trying to say. Support for what? Which case?

Reviewer: Something like this?

```suggestion
> [!TIP]
> Feedback for improving support for parsing and rendering doc-comments is highly appreciated.
> [Open an issue](https://github.com/NixOS/nixpkgs/issues/new?labels=6.topic%3A+documentation&title=Doc%3A+) to request bugfixes or new features.
```

Would be better if you created an issue template where you ask for what you need to know directly, then link it here.
